kb/deadline-exceeded-troubleshooting.md

```diff
@@ -0,0 +1,118 @@
+---
+slug: deadline-exceeded-troubleshooting
+title: DeadlineExceeded error troubleshooting
+tags:
+  - kb-article
+date: 2022-11-09T00:00:00Z
+---
+
+All client-side requests made to the [Temporal Cluster](/concepts/what-is-a-temporal-cluster) are [gRPC requests](https://grpc.io/docs/what-is-grpc/core-concepts/#deadlines).
+Sometimes, when these requests can't be completed, you'll see this particular error message: `Context: deadline exceeded`.
+Network interruptions, timeouts, server overload, and Query errors are some of the reasons that can cause this error.
+
+The following sections discuss the nature of this error and how to troubleshoot it.
+
+#### Check frontend service logs
+
+:::note
+
+Cloud users cannot access some of the logs needed to diagnose the source of the error.
+
+If you're using Temporal Cloud, check your Workflow History for errors and create a [support ticket](https://support.temporal.io/) with Temporal.
+Provide the full error message in your ticket.
+
+:::
+
+```
+{“level”:“error”,“ts”:“2022-03-21T19:32:42.312Z”,“msg”:“unavailable error”,“service”:“frontend”,“error”:“unable to get temporal-sys-add-search-attributes-workflow workflow state: context deadline exceeded”,“logging-call-at”:“adminHandler.go:1163”,“stacktrace”:“go.temporal.io/server/common/log.
+```
+
+Frontend service logs can show which parts of the Cluster aren't working.
+
+Verify that the Frontend Service is connected by opening the Web UI in your browser.
+
+Alternatively, OSS users can check that Frontend and other service are running with
+
+```
+tctl cluster health
+```
+
+and
+
+```
+alsovia grpcurl
+
+grpc-health-probe
+
+```
+
+- [History Service](/clusters#history-service)
+- [Persistence](/clusters#persistence) database
+- [Advanced Visibility](/visibility#advanced-visibility) database
+
+Logs can also be used to find Client [Query](/workflows#queries) requests that failed with this error.
+
+#### Check your Cluster metrics
+
+Cluster metrics can be used to detect issues (such as 'resource exhausted') that implact cluster health.
+This can cause your client to fail and send this error.
```

Comment:
what error? deadline exceeded or resource exhausted? maybe if you think is ok to specify that this could cause the deadline exceeded error in client code or tctl commands